osier/models/dispatch.py

```diff
@@ -78,30 +107,128 @@ class DispatchModel():
+    `None`. This is replaced by a setter method in :class:`Dispatch`.
+
+    3. :meth:`_write_model_equations` may be called before :meth:`solve` if
+    users wish to add their own constraints or parameters to the problem.
     """
 
     def __init__(self,
                  technology_list,
                  net_demand,
+                 time_delta=None,
+                 power_units=MW,
                  solver='cplex',
                  lower_bound=0.0,
                  oversupply=0.0,
                  undersupply=0.0):
         self.net_demand = net_demand
-        self.technology_list = technology_list
+        self.time_delta = time_delta
+        self.power_units = power_units
         self.lower_bound = lower_bound
         self.oversupply = oversupply
         self.undersupply = undersupply
         self.model = ConcreteModel()
         self.solver = solver
         self.results = None
         self.objective = None
+        self.model_initialized = False
+
+        self.technology_list = synchronize_units(
+            technology_list,
+            unit_power=power_units,
+            unit_time=self.time_delta.units)
+
+    @property
+    def time_delta(self):
+        return self._time_delta
+
+    @time_delta.setter
+    def time_delta(self, value):
+        if value:
+            valid_quantity = _validate_quantity(value, dimension='time')
+            self._time_delta = valid_quantity
+        else:
+            if isinstance(self.net_demand, pd.DataFrame):
+                try:
+                    freq_list = list(self.net_demand.index.inferred_freq)
+                    freq_key = freq_list[-1]
+                    try:
+                        value = float(freq_list[0])
+                    except ValueError:
+                        warnings.warn((f"Could not convert value "
+                                       f"{freq_list[0]} to float. "
+                                       "Setting to 1.0."),
+                                      UserWarning)
+                        value = 1.0
+                    self._time_delta = _validate_quantity(
+                        f"{value} {_freq_opts[freq_key]}", dimension='time')
+                except KeyError:
+                    warnings.warn(
+                        (f"Could not infer time delta with freq {freq_key} "
+                         "from pandas dataframe. Setting delta to 1 hour."),
+                        UserWarning)
+                    self._time_delta = 1 * hr
+            else:
+                self._time_delta = 1 * hr
 
     @property
-    def n_hours(self):
+    def power_units(self):
+        return self._demand_units
+
+    @power_units.setter
+    def power_units(self, value):
+        if value:
+            valid_quantity = _validate_unit(value, dimension='power')
+            self._demand_units = valid_quantity
+        else:
+            warnings.warn(f"Could not infer demand units. Unit set to MW.")
```

Comment:
I think there might be a missing `self._demant_units = 'MW'` missing here.

Reply:
Yes! Good catch

Reply:
Also, it should be `_power_units`. I was originally going to do demand units, but I think power units makes more sense.